### PR TITLE
Add backend tests and linting setup

### PR DESCRIPTION
**Pull request by 8bury**
Add unit testing framework with testify, strict golangci-lint configuration, and CI/CD workflow to ensure code quality and reliability.

- Add testify for assertions and mocking
- Create mock implementations for UserDAO and RefreshTokenDAO
- Implement 32 comprehensive unit tests for auth_service covering:
  * Registration with validation (username, email, password)
  * Login success and failure scenarios
  * Token refresh with all edge cases (expired, revoked, mismatched)
  * Profile updates and validations
  * Helper functions (validateUsername, validateEmail, etc.)
- Configure golangci-lint with 25+ strict linters
- Add Makefile with test, lint, and coverage commands
- Add GitHub Actions workflow for automated testing and linting
- All tests passing with 21.2% initial coverage on services

Testing can be run with: make test
Linting can be run with: make lint

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive authentication test suite covering registration, login, token refresh, logout, and profile updates.

* **Chores**
  * Introduced CI pipeline for testing, linting, and building; added lint config and developer Makefile; updated module dependencies.

* **Bug Fixes**
  * Improved runtime error handling to prevent panics (server start and JWT claim parsing).

* **Style**
  * Standardized and normalized user-facing error message casing.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: .github/workflows/backend-ci.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+      - name: Download dependencies
+        working-directory: backend
+        run: go mod download
+
+      - name: Run tests
+        working-directory: backend
+        run: go test ./... -race -coverprofile=coverage.out -covermode=atomic -timeout 30s
+
+      - name: Upload coverage to Codecov
```
Comment (8bury) [inline .github/workflows/backend-ci.yml:39]: @claude i dont want to use this codecov, delete it
